dashboard/test/lib/expired_deleted_account_purger_test.rb

```diff
@@ -0,0 +1,263 @@
+
+    found_accounts = ExpiredDeletedAccountPurger.new.send :soft_deleted_accounts
+
+    assert_includes found_accounts, soft_deleted_account
+    refute_includes found_accounts, active_account
+    refute_includes found_accounts, hard_deleted_account
+  end
+
+  test 'does not locate accounts that have not been soft-deleted' do
+    active_account = create :student
+    soft_deleted_account = create :student, deleted_at: 29.days.ago
+
+    picked_users = ExpiredDeletedAccountPurger.new(
+      deleted_after: 30.days.ago,
+      deleted_before: 28.days.ago
+    ).send :expired_deleted_accounts
+
+    assert_includes picked_users, soft_deleted_account
+    refute_includes picked_users, active_account
+  end
+
+  test 'does not locate accounts that have already been purged' do
+    purged_account = create :student, deleted_at: 29.days.ago, purged_at: 1.day.ago
+    unpurged_account = create :student, deleted_at: 29.days.ago
+
+    picked_users = ExpiredDeletedAccountPurger.new(
+      deleted_after: 30.days.ago,
+      deleted_before: 28.days.ago
+    ).send :expired_deleted_accounts
+
+    assert_includes picked_users, unpurged_account
+    refute_includes picked_users, purged_account
+  end
+
+  test 'does not locate accounts deleted before the start date' do
+    account_deleted_before_cutoff = create :student, deleted_at: 31.days.ago
+    account_deleted_after_cutoff = create :student, deleted_at: 29.days.ago
+
+    picked_users = ExpiredDeletedAccountPurger.new(
+      deleted_after: 30.days.ago,
+      deleted_before: 28.days.ago
+    ).send :expired_deleted_accounts
+
+    assert_includes picked_users, account_deleted_after_cutoff
+    refute_includes picked_users, account_deleted_before_cutoff
+  end
+
+  test 'does not locate accounts deleted less than 28 days ago' do
+    account_deleted_too_recently = create :student, deleted_at: 27.days.ago
+    account_deleted_long_enough = create :student, deleted_at: 29.days.ago
+
+    picked_users = ExpiredDeletedAccountPurger.new(
+      deleted_after: 30.days.ago,
+      deleted_before: 28.days.ago
+    ).send :expired_deleted_accounts
+
+    assert_includes picked_users, account_deleted_long_enough
+    refute_includes picked_users, account_deleted_too_recently
+  end
+
+  test 'locates only accounts within custom window' do
+    deleted_10_days_ago = create :student, deleted_at: 10.days.ago
+    deleted_20_days_ago = create :student, deleted_at: 20.days.ago
+    deleted_30_days_ago = create :student, deleted_at: 30.days.ago
+
+    picked_users = ExpiredDeletedAccountPurger.new(
+      deleted_after: 25.days.ago,
+      deleted_before: 15.days.ago
+    ).send :expired_deleted_accounts
+
+    refute_includes picked_users, deleted_10_days_ago
+    assert_includes picked_users, deleted_20_days_ago
+    refute_includes picked_users, deleted_30_days_ago
+  end
+
+  test 'does not locate any accounts when window is negative' do
```

Comment:
Should we perhaps raise an error when the window is negative or 0?

Reply:
No; this is actually the expected behavior of the script until August 28th.  It's reasonable for the script to find no matching accounts and quit in this case.